src/graphql/schemas/typeDefs/signatureRequestTypeDefs.ts

```diff
@@ -0,0 +1,64 @@
+import { Field, Int, ObjectType, registerEnumType } from "type-graphql";
+
+enum SignatureRequestPurpose {
+  UPDATE_USER_DATA = "update_user_data",
+}
+
+enum SignatureRequestStatus {
+  PENDING = "pending",
+  EXECUTED = "executed",
+  CANCELED = "canceled",
+}
+
+registerEnumType(SignatureRequestPurpose, {
+  name: "SignatureRequestPurpose",
+  description: "Purpose of the signature request",
+});
+
+registerEnumType(SignatureRequestStatus, {
+  name: "SignatureRequestStatus",
+  description: "Status of the signature request",
+});
+
+@ObjectType({
+  description: "Pending signature request for a user",
+  simpleResolvers: true,
+})
+class SignatureRequest {
+  @Field({
+    description: "The safe address of the user who needs to sign",
+  })
+  safe_address?: string;
+
+  @Field({
+    description: "The hash of the Safe message (not the message to be signed)",
+  })
+  message_hash?: string;
+
+  @Field(() => String, {
+    description: "The message data in JSON format",
+  })
+  message?: string;
+
+  @Field({
+    description: "When the signature request was created",
+  })
+  created_at?: string;
```

Comment:
See earlier comment on harmonising the datatypes, which means we have a seconds timestamp.

Reply:
Just to make sure I understand correctly: We want to change this to `number` as that's how it's represented internally and this would be the first time we're doing it this way, right? I've had a look at the other type definitions and none of the timestamps there are of type `number`.

Reply:
For reference from our chat: 

It should be `numeric(78,0)` which is the type that supports the `bigint` value range. As a rule of thumb on determining the type of on-chain data in off-chain representation, in case nobody else is online, check against what the value is on-chain and match that as closely as possible. Most on chain numbers are `uint256` in our context, which is `bigint`

Reply:
Now you're mixing a few different things, though. Timestamps like this one (`created_at`, `updated_at`, etc.) are stored in `timestamptz` as far as I can tell from looking at our database. They default to the PG function `now()` if they don't get an explicit value. Internally those are stored as ISO 8601 strings if I'm not mistaken. I think your comment in chat was about me asking about the chain ids and, yes, the same would apply to timestamps taken from the chain. But this is no such case.

I had a look at our code and couldn't immediately see any other case where we expose a Postgres timestamp through GraphQL, so this is a bit of a precedent and since this is not on-chain data, I couldn't apply your rule of thumb. 

So, given your comments here and in Discord I think what you're looking for is to have all dates and timestamps that are exposed via GraphQL to be in the same format, no matter where they're coming from. The desired format is Unix timestamp in seconds as a string. Please let me know if I'm missing something :)